okcupyd/profile.py

```diff
@@ -213,14 +215,25 @@ def essays(self):
         """
         return essay.Essays(self)
 
+    _age_xpb = xpb.span.with_class('userinfo2015-basics-asl-age')
+    _user_age_xpb = xpb.span(id='ajax_age')
+
     @util.cached_property
     def age(self):
         """
         :returns: The age of the user associated with this profile.
         """
-        return int(xpb.span(id='ajax_age').get_text_(self.profile_tree).strip())
 
-    _percentages_and_ratings_xpb = xpb.div(id='percentages_and_ratings')
+        try:
+            #Attempt to get a non logged-in user's profile age
+            profile_age = int(self._age_xpb.get_text_(self.profile_tree))
+        except:
```

Comment:
rewrote more elegantly, using self.is_logged_in_user, in https://github.com/kfred/okcupyd/commit/da49f95d1fdeddd57f96d82ced2050cd32c29bcb